File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    done = [False] * len(progresses)
    n = len(progresses)
    k = 0
    cnt = 0
    while not all(done):
        for i in range(n):
            progresses[i] += speeds[i]
            if progresses[i] >= 100:
                done[i] = True
                progresses[i] = 100
        
        if done[k]:
            cnt = 0

            while k < n and done[k]:
                cnt += 1
                k += 1

            answer.append(cnt)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <vector>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    int n = progresses.size();
    vector<bool> done(n, false);
    
    int k = 0;

    while (k < n) {
        for (int i = 0; i < n; i++) {
            progresses[i] += speeds[i];

            if (progresses[i] >= 100) {
                done[i] = true;
                progresses[i] = 100;
            }
        }

        if (done[k]) {
            int cnt = 0;

            while (k < n && done[k]) {
                cnt++;
                k++;
            }

            answer.push_back(cnt);
        }
    }

    return answer;
}